tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,317 @@
+    return filtered_dataset
+
+
+def filter_code(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Filters the given dataset based on various criteria.
+    Refer to the full list of all filters here:
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/config/heuristic_filter_en.yaml
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/tutorials/peft-curation/main.py
+
+    Args:
+        dataset (DocumentDataset): The dataset to be filtered.
+
+    Returns:
+        DocumentDataset: The filtered dataset.
+    """
+    filters = Sequential(
+        [
+            # If a document contains a number of words not
+            # within a specified range then discard
+            ScoreFilter(
+                WordCountFilter(min_words=5, max_words=100000),
+                text_field="text",
+                score_field="word_count",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+def filter_code_dataset(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Filters the given code dataset based on various code-specific criteria.
+    Refer to the full list of all filters here:
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/config/heuristic_filter_code.yaml
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/filters/code.py
+
+    Args:
+        dataset (DocumentDataset): The dataset to be filtered.
+
+    Returns:
+        DocumentDataset: The filtered dataset.
+    """
+    filters = Sequential(
+        [
+            # if the comment to code ratio is not within the specified range,
+            # discard the document
+            ScoreFilter(
+                PythonCommentToCodeFilter(
+                    min_comment_to_code_ratio=0.001, max_comment_to_code_ratio=0.80
+                ),
+                text_field="text",
+                score_type=float,
+            ),
+            # if the number of lines of code is not within the specified range,
+            # discard the document
+            ScoreFilter(
+                NumberOfLinesOfCodeFilter(min_lines=5, max_lines=20000),
+                text_field="text",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+# +
+def redact_pii(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redacts personally identifiable information (PII) from extracted comment lines in a given dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents with PII.
+
+    Returns:
+        DocumentDataset: The redacted dataset with PII replaced by a generic value.
+    """
+    DEFAULT_MAX_DOC_SIZE = 3000000
+    redactor = Modify(
+        PiiModifier(
+            supported_entities=[
+                "PERSON",
+                "EMAIL_ADDRESS",
+            ],
+            anonymize_action="replace",
+            device="gpu",
```

Comment:
Question: If we specify device="gpu" here will this work on cpu only systems?

Reply:
Yes this still works only on cpu. looks like its defaulting to cpu if not available.